src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -801,7 +861,23 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
 
     proxy_cert = proxy_cert.replace('\\', r'\\\\')
 
-    if https_proxy == "" and http_proxy == "" and no_proxy == "" and proxy_cert == "" and not disable_proxy and not auto_upgrade:
+    # Prompt if private link is getting enabled
+    if enable_private_link == "true":
+        if os.getenv('SKIP_PROMPT') != "true":
+            if not prompt_y_n("The Cluster Connect and Custom Location features are not supported by Private Link at this time. Enabling Private Link will disable these features. Are you sure you want to continue?"):
+                return
+
+    # Set preview client if private link properties are provided.
+    preview_sdk = False
+    connected_cluster = get_connectedk8s(cmd, client, resource_group_name, cluster_name)
+    if enable_private_link or connected_cluster.private_link_state.lower() == "enabled":
+        client = cf_connected_cluster_prev_2021_04_01(cmd.cli_ctx, None)
+        preview_sdk = True
+
+    # Patching the connected cluster ARM resource
+    patch_cc_response = update_connected_cluster_internal(client, resource_group_name, cluster_name, tags, enable_private_link, private_link_scope_resource_id, no_wait, preview_sdk)
+
+    if https_proxy == "" and http_proxy == "" and no_proxy == "" and proxy_cert == "" and not disable_proxy and not auto_upgrade and not tags and not enable_private_link:
```

Comment:
This statement has too many judgment conditions. Could you divide it into several categories and judge them in turn to increase the readability of this code?

Reply:
Yes, I understand. I mean, it's not a logical separation, it's just a separation of code style
Such as:
```Python
xxx_proxy = (https_proxy == "" and http_proxy == "" and no_proxy == "" and proxy_cert == "")
if xxx_proxy and not disable_proxy and not auto_upgrade and not tags and not enable_private_link:
```

Reply:
got it, updated.